crates/compiler/can/src/def.rs

```diff
@@ -932,7 +941,12 @@ pub(crate) fn canonicalize_defs<'a>(
     scope: &mut Scope,
     loc_defs: &'a mut roc_parse::ast::Defs<'a>,
     pattern_type: PatternType,
-) -> (CanDefs, Output, MutMap<Symbol, Region>) {
+) -> (
```

Comment:
This isn't actionable for your change, but I think the style of APIs in this file could be improved over time. A lot of the functions both take in values that are mutated through a pointer, and output values that are consumed. I wonder if it would be easier to follow if they went strongly one way or another; i.e. only using a mutable `env`, or only outputing immutable contexts that the consumers then compose.